libsel4arm-vmm/src/arch/arm/devices/vgic.c

```diff
@@ -297,11 +297,11 @@ static int vgic_vcpu_inject_irq(struct device *d, vm_t *vm, struct virq_handle *
         return err;
     } else {
         /* Add to overflow list */
-        int idx = LR_OF_NEXT(vgic->lr_overflow.tail);
-        ZF_LOGF_IF(idx == vgic->lr_overflow.head && idx != vgic->lr_overflow.tail,
-                   "too many overflow irqs");
+        int idx = vgic->lr_overflow.tail;
+        ZF_LOGF_IF(idx == vgic->lr_overflow.head &&
+                   vgic->lr_overflow.head != vgic->lr_overflow.tail, "too many overflow irqs");
```

Comment:
Why is it an error if the head does not equal the tail? Also if the first condition is true the second must be false?

Reply:
When `head == tail` the buffer is empty, so we can allocated the head in this case.

Reply:
otherwise the buffer is full and we're about to overwrite the current head

Reply:
Thanks for the fix!